pgtype/timestamp.go

```diff
@@ -116,17 +116,21 @@ func (ts *Timestamp) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
-type TimestampCodec struct{}
+type TimestampCodec struct {
+	// ScanLocation is the location that the time is assumed to be in for scanning. This is different from
+	// TimestamptzCodec.ScanLocation in that this setting does change the instant in time that the timestamp represents.
+	ScanLocation *time.Location
+}
 
-func (TimestampCodec) FormatSupported(format int16) bool {
+func (*TimestampCodec) FormatSupported(format int16) bool {
```

Comment:
Question: Is it strictly necessary for the new codec implementations to use a pointer receiver? In both cases the only member is the `*time.Location` field which is itself a pointer. As you mentioned in your commit descriptions, this is technically a breaking change, albeit a very minor one.

Reply:
Is it strictly necessary? No. But as I understand it, it is usually better for interfaces implemented by structs to use methods that receive the pointer rather than the value.